src/Microsoft.AspNetCore.Antiforgery/DefaultClaimUidExtractor.cs

```diff
@@ -57,25 +63,29 @@ internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity
             return identifierParameters;
         }
 
-        private static byte[] ComputeSHA256(IEnumerable<string> parameters)
+        private byte[] ComputeSha256(IEnumerable<string> parameters)
         {
-            using (var stream = new MemoryStream())
+            var serializationContext = _pool.Get();
+
+            try
             {
-                using (var writer = new BinaryWriter(stream))
+                var writer = serializationContext.Writer;
+                foreach (string parameter in parameters)
                 {
-                    foreach (string parameter in parameters)
-                    {
-                        writer.Write(parameter); // also writes the length as a prefix; unambiguous
-                    }
+                    writer.Write(parameter); // also writes the length as a prefix; unambiguous
+                }
 
-                    writer.Flush();
+                writer.Flush();
 
-                    using (var sha256 = SHA256.Create())
-                    {
-                        var bytes = sha256.ComputeHash(stream.ToArray(), 0, checked((int)stream.Length));
-                        return bytes;
-                    }
-                }
+                var sha256 = serializationContext.Sha256;
+                var stream = serializationContext.Memory;
+                var bytes = sha256.ComputeHash(stream.ToArray(), 0, checked((int)stream.Length));
```

Comment:
:see_no_evil: for now. Will reevaluate various `ToArray()` and `ToString()` calls later.